indico/modules/events/timetable/views/__init__.py

```diff
@@ -129,4 +129,6 @@ def inject_meeting_body(event, **kwargs):
 
 def _entry_title_key(entry):
     obj = entry.object
-    return obj.full_title if entry.type == TimetableEntryType.SESSION_BLOCK else obj.title
+    if entry.type == TimetableEntryType.SESSION_BLOCK:
+        return (obj.session.code, obj.full_title)
```

Comment:
Do we want the first element at all in case there is no code?

Reply:
What if there is one entry with a code and one without? We'd be comparing the code against the title which is not ideal. Alternatively, we could return a 2-tuple like `('', obj.title)` for non-block entries. That way, we always compare the corresponding attributes.

Reply:
I was more thinking of cases where a contribution and a session start at the same time, and neither use codes. In that case there should be alphabetic sorting by title IMHO.

But yes, always returning a 2-tuple would work and be a pretty clean solution.

Reply:
pushed, should be g2g